packages/rum-core/src/domain/rumEventsCollection/resource/resourceCollection.ts

```diff
@@ -231,9 +231,14 @@ function computeIndexingInfo(sessionManager: RumSessionManager, resourceStart: C
 }
 
 function computePageStateInfo(pageStateHistory: PageStateHistory, startClocks: ClocksState, duration: Duration) {
+  if (!isExperimentalFeatureEnabled('resource_page_states')) {
+    return
+  }
+
   return {
     _dd: {
       page_states: pageStateHistory.findAll(startClocks.relative, duration),
+      page_was_discarded: (document as any).wasDiscarded,
```

Comment:
<!-- Suggestions are specific requests to improve the subject under review. It is assumed that we all want to do what's best, so these comments are never dismissed as “mere suggestions”, but are taken seriously.  -->
**💬 suggestion:** ‏maybe `String((document as any).wasDiscarded)` so you can differentiate between true/false/undefined?

Reply:
I don't see the point of getting undefined. true and false should be okay but maybe I'm missing something?

Reply:
Not all browsers are supporting this API, so when it doesn't, the `page_was_discarded` property would be missing, and it would be a bit less trivial to know whether this property is missing because an older SDK has been used, because the FF was not enabled or because the browser don't support it.

That's why I think it might help if you had an explicit `undefined` value here.